### ISSUE TITLE
"ESLint couldn't find the plugin" notice loses a last character

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#13255]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 12.0.0
* **npm Version:** 6.14.4

**What parser (default, Babel-ESLint, etc.) are you using?**

Not relevant, but `"@typescript-eslint/parser"`

**Please show your full configuration:** 

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: [
    "plugin:eslint-config-prettier",
  ],
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
eslint --print-config ./eslintrc.js
```

**What did you expect to happen?**

The module name being reported as not existing should be ... `eslint-config-prettier`, I'm guessing?

**What actually happened? Please include the actual, raw output from ESLint.**

```
Oops! Something went wrong! :(

ESLint: 6.8.0.

ESLint couldn't find the plugin "eslint-plugin-eslint-config-prettie".

(The package "eslint-plugin-eslint-config-prettie" was not found when loaded as a Node module from the directory "C:\Code\tslinttest".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-eslint-config-prettie@latest --save-dev

The plugin "eslint-plugin-eslint-config-prettie" was referenced from the config file in ".eslintrc.js".

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

This issue split from: https://github.com/typescript-eslint/tslint-to-eslint-config/issues/439.

**Are you willing to submit a pull request to fix this bug?**

I'd love to! 🤗

**Comment by yeonjuan** [comment]
Hi @JoshuaKGoldberg  :) Thanks for the reporting.  😀

It seems a bug to me; I agree that the error message should be improved

> The module name being reported as not existing should be ... eslint-config-prettier, I'm guessing?

Maybe not, 🤔 because the actual reason is incorrect configuration.

```
"plugin:eslint-config-prettier",
```

The "plugin:xxx" means the use of exported configuration from other plugins.
so it should be consists of  `"plugin:"` + `{package name}` + `"/"` + `{configuration name}`
- [Using the configuration from a plugin](https://eslint.org/docs/user-guide/configuring#using-the-configuration-from-a-plugin)
> The extends property value can consist of:
> * `plugin:`
> * the package name (from which you can omit the prefix, for example, `react`)
>  * `/`
> * the configuration name (for example, `recommended`)


---

https://github.com/eslint/eslint/blob/015edf6467e33c67b904db037a674d71957a6865/lib/cli-engine/config-array-factory.js#L804

the problem is here. there is no '/' character in `extendName ` so `lastIndexOf ` return `-1`;

**Comment by Vovch** [comment]
Hi, everyone! 

Stumbled upon this issue in other person's code yesterday.

As a solution I might propose adding the next code:

```
if (slashIndex === -1) {
    const wrongConfigName = JSON.stringify(extendName).slice(1, -1);
    throw new Error(`'extends' requires a configuration name for '${wrongConfigName}', provided after '/' sign.`);
}
```

This way it is possible to also give an understandable error in some cases when the user mistakenly prints `\` sign. E.g.:
`Error: 'extends' requires a configuration name for 'plugin:prettier\recommended', provided after '/' sign.`

instead of:
`ecommende". because there is whitespace in the name. Please check your configuration and remove all whitespace from the plugin name.`

**Comment by mysticatea** [comment]
Thank you for your report.

But it's not a bug. The `plugin:` specifier needs a plugin name and a config name: `plugin:foo/bar`. Here, the `foo` is a plugin name, it means loading `eslint-plugin-foo`, and the `bar` is a config name, it means using the `require("eslint-plugin-foo").configs.foo` config object.

**Comment by jaganathanb** [comment]
I faced same kind of issue but for `@typescript-eslint/eslint-plugin`. The surprising thing is that the issue is because of `eslint-nullish-coalescing` which changed the content of `node_modules\.bin\eslint.cmd` file to some thing like below.

```
@IF EXIST "%~dp0\node.exe" (
  "%~dp0\node.exe"  "%~dp0\..\eslint-nullish-coalescing\bin\eslint.js" %*
) ELSE (
  @SETLOCAL
  @SET PATHEXT=%PATHEXT:;.JS;=;%
  node  "%~dp0\..\eslint-nullish-coalescing\bin\eslint.js" %*
)
```
I installed eslint version 7 but the error message was showing up as 4.19.1 which the `eslint-nullish-coalescing` package version which is a fork of eslint. 

Please check your eslint.cmd file's content.

[Here](https://stackoverflow.com/a/62006428/744534) is the stackoverflow post.

**Comment by cellog** [comment]
this is working as intende? That seems like a bad desig. Tell people they used invalid syntax, don't just chop off the last characte.

**Comment by mdjermanovic** [comment]
PR to print an appropriate error message in this case: #13473

**Comment by mdjermanovic** [comment]
Fixed by #13473.
